### PR TITLE
Fixes to connection typings

### LINKED ISSUE #11983: types: wrong types related to mongoose.connection
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Mongoose version

6.4.0

### Node.js version

16

### MongoDB server version

5

### Description

After upgrading a few mongoose versions, some of our typings broke.

See the repro, add those lines to `connection.test.ts`

### Steps to Reproduce

Add these lines to `test/types/connection.test.ts`. We use this in unit tests and the model isn't important.

```ts
expectType<Promise<void>>(conn.model('test').ensureIndexes()); // Object is of type 'unknown'.ts(2571)

expectType<Promise<mongodb.InsertManyResult<AnyObject>>>(
  conn.collection('test').insertMany([{ _id: 'string' }]) // Type 'string' is not assignable to type 'ObjectId'.ts(2322)
);
```

### Expected Behavior

Both should be allowed.

### PR DESCRIPTION
**Pull request by andreialecu**
**Summary**

Fixes #11983

### CODE REVIEW

File: types/index.d.ts [hunk 1]
```diff
@@ -108,6 +108,8 @@ declare module 'mongoose' {
     [k: string]: any
   }
 
+  export type AnyObjectAnyId = AnyObject & { _id?: string | number | boolean | object };
```
Comment (vkarpov15) [inline types/index.d.ts:111]: Why is this necessary? Can't we do `_id?: any` ?

Comment (andreialecu) [reply types/index.d.ts:111]: `{ _id?: any }` doesn't work; it was what I initially tried. There's a conditional type somewhere that doesn't agree with that. :)

And `AnyObject` should already include `{ _id?: any }` implicitly by how it's defined:
```
  export interface AnyObject {
    [k: string]: any
  }
```

You can try replacing it with `_id?: any`; you can see that the error comes back.

File: test/types/connection.test.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import { createConnection, Schema, Collection, Connection, ConnectionSyncIndexesResult, Model } from 'mongoose';
+import { createConnection, Schema, Collection, Connection, ConnectionSyncIndexesResult, Model, AnyObject, AnyObjectAnyId } from 'mongoose';
```
Comment (andreialecu) [inline test/types/connection.test.ts:1]:
```suggestion
import { createConnection, Schema, Collection, Connection, ConnectionSyncIndexesResult, Model, AnyObjectAnyId } from 'mongoose';
```
